### ISSUE TITLE
Export Image Support for iOS

### ISSUE THREAD

**Issue by michaelsboost** [photopea/photopea#1166]
I just tried using Photopea on my iPhone. Works great except I cannot save the image 😞 it opens a new tab with about: as the domain but doesn’t show anything. 

Any plans to resolve this issue?

**Comment by photopea** [comment]
Hi, I think that your browser does not allow any files to be stored. Have you ever downloaded anything in your mobile browser (like some file) ? Or if you could show us a website, that lets you download a file in your brwoser, we would do it the same way as that website.

**Comment by michaelsboost** [comment]
Yeah I've made a few apps where I can download images like my [Flat Design Character Maker](https://github.com/michaelsboost/Flat-Design-Character-Maker) or the [Cartoon Avatar Creator](http://michaelsboost.github.io/Cartoon-Avatar-Creator). 

If it is something like a zip file though then the zip file MUST be hosted on a server as it will not work with FileReader. (which you can read more about [here](https://stackoverflow.com/a/51072795))

**Comment by photopea** [comment]
I am very sorry, I have no iOS device to be able to test it :(

But according to this, saving files works well in iOS: https://www.youtube.com/watch?v=w1k0_z87Wbs  So what exactly happens when you try to save in your iOS device?

**Comment by photopea** [comment]
Photopea uses the "download" attribute, which is currently not available only in iOS Safari and Interent Explorer :(  https://caniuse.com/#feat=download

**Comment by michaelsboost** [comment]
Roger that! After some research, I found that the reason the download attribute works on desktop and android but not iOS is because iOS doesn't have a file system. The alternative is to open the file on the browser with the proper mime type, so it can show its content (and the user can then manually download it if needed).  
[source 1](https://stackoverflow.com/a/57144390/710887) [source 2](https://bugs.webkit.org/show_bug.cgi?id=167341) [source 3](https://github.com/eligrey/FileSaver.js/issues/375#issuecomment-341518668)

    var img = new Image();
    img.crossOrigin = "Anonymous";
    img.id = "getshot";
    img.src = imageURL;
    document.body.appendChild(img);

    var a = document.createElement("a");
    a.href = getshot.src;
    a.download = "workout_log " + dateTime + " " + startTime + ".png";
    a.click();
    document.body.removeChild(img);

**Comment by BRodrigues98** [comment]
Hi! Letting you know I have faced the same issue on iOS 16.3 with an iPhone XR. Leaving this here as I have used a workaround which might help others.

The workaround that worked for me was to share it to Imgur, and then saving from there.

**Comment by photopea** [comment]
I just tried to use www.Photopea.com on my iPad (iOS 15.5) in Chrome and Safari. I tried to press File - Export - JPG. Images were exported in both cases :(

I will download iOS 16.3 and see how it works.

**Comment by photopea** [comment]
I have downloaded the latest iOS 16.3.1 to my iPad. Exporting a JPG still works. It is saved "On My iPad" by Chrome, and onto the "iCloud Drive" by Safari.
